src/main/java/seedu/address/model/assignment/Assignment.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.model.assignment;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents an Assignment in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Assignment {
+
+    private final Description description;
+    private final DueDate dueDate;
+    private final Status status;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Assignment(Description description, DueDate dueDate, Status status) {
+        requireAllNonNull(description, dueDate, status);
+        this.description = description;
+        this.dueDate = dueDate;
+        this.status = status;
+    }
+
+    public Description getDescription() {
+        return description;
+    }
+
+    public DueDate getDueDate() {
+        return dueDate;
+    }
+
+    public Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Returns true if both persons have the same name.
+     * This defines a weaker notion of equality between two persons.
+     */
+    public boolean isSameAssignment(Assignment otherAssignment) {
+        if (otherAssignment == this) {
+            return true;
+        }
+
+        return otherAssignment != null
+                && otherAssignment.getDescription().equals(getDescription());
+    }
+
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Assignment)) {
+            return false;
+        }
+
+        Assignment otherAssignment = (Assignment) other;
+        return otherAssignment.getDescription().equals(getDescription())
+                && otherAssignment.getDueDate().equals(getDueDate())
+                && otherAssignment.getStatus().equals(getStatus());
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(description, dueDate, status);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Perhaps the `toString()` method should segregate the different fields such as description and status with a newline instead of a `;` for better clarity to the users, unless the `;` will be used as delimiters to add in the newline when displaying happens?

Reply:
For ```toString()``` method, I used the same formatting as the ```toString()``` method in the ```Person``` class for consistency. 

As for improving the display message to the user, because we haven't discuss about it so I stick to the above template for now.